### ISSUE TITLE
Run fula under a android forgrand service

### ISSUE THREAD

**Issue by farhoud** [functionland/react-native-fula#52]
on android wrap the FULA client inside [service](https://developer.android.com/guide/components/foreground-services)
is the better way to run FULA clients this way:
- the service can run on its own process
- auto start on boot
- the service can create and control notification

If you need I can make PR. I have done it on a [project](https://github.com/hood-chat/app) like this with a more complicated interface.

**Comment by ehsan6sha** [comment]
@farhoud perfect, a PR for this would be great

**Comment by farhoud** [comment]
https://github.com/functionland/react-native-fula/pull/54
I started the PR who is working on this currently? @ehsan6sha

**Comment by ehsan6sha** [comment]
> #54 I started the PR who is working on this currently? @ehsan6sha

Thank you Farhood. I am the one working on it

**Comment by farhoud** [comment]
So there are 2 services fula and wnfs.
Do they need to run in background?
Or they are schedule jobs?
Or non they only needed when the app is open?

**Comment by farhoud** [comment]
Let me share my reason for these question's.
Android is limiting background service and added some new task management APIs.
There are type of service's which run on its own process and use bridge interface to communicate with activities. So task like streaming data from ui are not suitable for these kind of service's.

And another ting do you need android native notification?

**Comment by ehsan6sha** [comment]
They are not scheduled jobs. WNFS is responsible for encryption and creating merkle dag from a file, amd fula transfers the created dag to blox. Also whenever user download a file, wnfs requests the needed cids and fula bring them to wnfs from blox and wnfs decrypt .

So to the question:

1- i think if we keep libp2p connection always open, it drains battery and we need to turn it on when application is in foreground (open) and when app is in background or closed, schedule it to turn on and check for updates and then close the connection 
2- we need notification which will be on top of fula libp2p but we don't have a notification service yet to blox and back

Did this answer all questions?

**Comment by farhoud** [comment]
My situation in Hoodchat was pretty much different and I need libp2p to always run. but based on [this Documentation](https://developer.android.com/about/versions/oreo/background) I think we can implement upload and download as Jobs and let the android job scheduler handle them:
- https://developer.android.com/about/versions/oreo/android-8.0#jobscheduler
this way we can also fix this #29 

but bounded service and foreground service are still an option.
So @ehsan6sha What do you think we should do?

**Comment by ehsan6sha** [comment]
Doesn't always running libp2p connection means keeping it open and drains battery?

**Comment by farhoud** [comment]
Sure big time especially if I enable DHT and hole punching but that's the price of privacy.
I don't have a choice I'm waiting on chats, but you have multiple choices.
